docs-ui/stories/components/compactSelect.stories.js

```diff
@@ -92,3 +83,35 @@ _CompactSelect.argTypes = {
   maxMenuHeight: {control: {type: 'number'}},
   triggerLabel: {control: {type: 'text'}},
 };
+
+export const _CompactSelectGrouped = props => (
+  <CompactSelect
+    defaultValue={props.multiple ? ['choice_one'] : 'choice_one'}
+    options={[
+      {
+        label: 'Section 1',
+        showToggleAllButton: true,
+        options: [
+          {value: 'choice_one', label: 'Choice One'},
+          {value: 'choice_two', label: 'Choice Two'},
+        ],
+      },
+      {
+        label: 'Section 2',
+        showToggleAllButton: true,
+        options: [
+          {value: 'choice_three', label: 'Choice Three'},
+          {value: 'choice_four', label: 'Choice Four'},
+        ],
+      },
+    ]}
+    {...props}
+  />
+);
```

Comment:
Add a "Grouped" story to better document the differences between `CompactSelect`s with grouped options vs those with a linear array of options.